active-rfcs/0000-portals.md

```diff
@@ -0,0 +1,321 @@
+
+```js
+`Teleport content could not be mounted to element with selector '${props.to}': element not found.
+
+// following would be a display where in the component tree this happened etc.
+```
+
+#### \$parent
+
+If the children of `<teleport>` contain any components, their `this.$parent` property should reference the `<teleport>`'s parent component. In other words, these components stay in their original spot in the _component tree_, even though they ended up mounted somewhere else in the _DOM tree_.
+
+`<teleport>`, not being a real component at all, is transparent and will not appear as an ancestor in the `$parent`chain.
+
+```html
+<template>
+  <teleport v-bind:to="targetName">
+    <Child />
+  </teleport>
+</template>
+<script>
+  export default {
+    name: 'Parent'
+    components: {
+      Child: {
+        template: '<div/>',
+        mounted() {
+          console.log(this.$parent.$options.name )
+          // => 'Parent'
+        }
+      }
+    },
+  }
+</script>
+```
+
+Similarly, using `inject` in `Child` should be able to inject any provided content from `Paren` or one of its ancestors.
+
+### Updating
+
+The `to` prop can be changed dynamically with `v-bind`. When the value changes, `<teleport>` will remove the children from the previous target and move them to the new one.
+
+If the children contain any component instances, these will not be influenced by this. The instances will be kept alive, keep their state etc.
+
+```html
+<template>
+  <teleport v-bind:to="targetName">
+    <p>This can be moved around with the button below</p>
+  </teleport>
+  <button v-on:click="toggleTarget">Toggle</button>
+  <hr />
+  <div id="A"></div>
+  <div id="B"></div>
+</template>
+<script>
+  export default {
+    data: () => ({
+      targetName: "A"
+    }),
+    methods: {
+      toggleTarget() {
+        this.targetName = this.targetName == "A" ? "B" : "A";
+      }
+    }
+  };
+</script>
+```
+
+If the new target selector doesn't match any elements:
+
+1. a warning should be logged during development.
+2. The content would stay mounted to the previous target element.
+
+### Destruction
+
+When a `<teleport>` is being destroyed (e.g. because its parent component is being destroyed or because of a `v-if`), its children are removed from the DOM and any component instances destroyed just like they were still children iof the parent.
+
+## Miscellaneous
+
+### Naming conflict with native portals
+
+the component introduced by this RFC was named `<portal>` in an earlier version of this RFC. But there's proposal for native portals:
+
+- Spec: https://wicg.github.io/portals/
+- Introduction: https://web.dev/hands-on-portals/
+
+Sinc we don't want to have a naming conflict with a future HTML element that may be called `<teleport>`, especially since it's functionality is about something completely different form what portals in libs like Vue or React mean right now, we chose to rename the component to `<teleport>`
```

Comment:
```suggestion
Sinc we don't want to have a naming conflict with a future HTML element that may be called `<portal>`, especially since it's functionality is about something completely different form what portals in libs like Vue or React mean right now, we chose to rename the component to `<teleport>`
```